src/app/CodeRepo/lib/types/revision.js

```diff
@@ -119,9 +119,18 @@ class Revision extends Type {
     }
 
     async skipReview() {
-        this.tags = this.tags.filter((tag) => !tag.startsWith("review:"));
+        await this.clearTags("review:", false)
         this.tags.push(`review:skip`);
-        await this.save()
+        await this.save();
+    }
+
+    async clearTags(pattern, save=true) {
+        this.tags = this.tags.filter((tag) => !tag.startsWith(pattern));
```

Comment:
All changes to type objects should be "inplace" not replaced, thus splice must be used or we might end up with issues when different async methods work with the same object at in parallel